### ISSUE TITLE
type 'ListTypeNode' is not a subtype of type 'NamedTypeNode' in type cast

### ISSUE THREAD

**Issue by SwannLV** [comigor/artemis#126]
Hello! 

I have this error when building: ` type 'ListTypeNode' is not a subtype of type 'NamedTypeNode' in type cast`

This is caused by this field in my schema.gql: 
`myField: [[ID]]`

Thank you 🙏🙂

**Comment by vasilich6107** [comment]
Could you provide complete shema sample and your query?

**Comment by SwannLV** [comment]
Sure: [build-error.zip](https://github.com/comigor/artemis/files/4580466/build-error.zip)

'myField' is the problem.

- schema.gql:
```
type Query {
  countries: [Country]

  search(filters: MissionSearchInput!): Items
}

input Input {
  myField: [[ID]]
}

type Items {
  id: String
  count: Int
}

type Country {
  id: String!
  iso2: String!
  iso3: String!
  name: String!
  phonePrefix: String!
}
```

- query.gql :
```
query MyQuery {
  countries{
    id
  }
} 
```

- build.yaml:
```
targets:
  $default:
    builders:
      artemis:
        options:
          generate_helpers: true
          fragments_glob: lib/**fragment.graphql
          schema_mapping:
            - queries_glob: lib/query.gql
              schema: lib/schema.graphql
              output: lib/graphql/types.dart
```

**Comment by comigor** [comment]
Hmm, I don't think we have a test for `List<List<Scalar>>` inputs, I'll take a look!

**Comment by SwannLV** [comment]
Hello! Would you know when this could be solved?
Thanks 🙏

**Comment by comigor** [comment]
@SwannLV This should be an easy fix, sorry for not looking at it early :(

But you'll have to bump past those latest breaking changes

**Comment by SwannLV** [comment]
Ok thanks a lot. Where could I see the list of those breaking changes?

**Comment by comigor** [comment]
On [CHANGELOG](https://github.com/comigor/artemis/blob/beta/CHANGELOG.md). I'm merging the fix right now.

**Comment by SwannLV** [comment]
Thanks. Is auto __typename optional?

**Comment by comigor** [comment]
The current changes only rename fields starting with `_` to `$`, typename will only  be converted when they're used (like into a inline spread)

**Comment by comigor** [comment]
Released as v6.3.2-beta.1 on pub.dev.
Feel free to reopen this if necessary!
